packages/commons/src/fhir/transformers/index.ts

```diff
@@ -974,9 +974,9 @@ function setResourceIdentifier(
     value: fieldValue
   } as TaskIdentifier
 
-  resource.identifier
+  resource.identifier = resource.identifier
     .filter((obj) => obj.system !== identifier.system)
-    .push(identifier)
+    .concat(identifier)
```

Comment:
This is where the bug was. `Array.prototype.filter` creates a new array instance so the `push` doesn't mutate the original array.